GameDataParser/Parsers/DungeonParser.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Xml;
+using GameDataParser.Crypto.Common;
+using GameDataParser.Files;
+using Maple2Storage.Types.Metadata;
+using System;
+
+
+namespace GameDataParser.Parsers
+{
+    public class DungeonParser : Exporter<List<DungeonMetadata>>
+    {
+        public DungeonParser(MetadataResources resources) : base(resources, "dungeon") { }
+
+        protected override List<DungeonMetadata> Parse()
+        {
+            List<DungeonMetadata> dungeons = new List<DungeonMetadata>();
+            foreach (PackFileEntry entry in Resources.XmlFiles)
+            {
+                if (!entry.Name.StartsWith("table/na/dungeonroom"))
+                {
+                    continue;
+                }
+
+                XmlDocument document = Resources.XmlMemFile.GetDocument(entry.FileHeader);
+                foreach (XmlNode dungeonNode in document.DocumentElement.ChildNodes)
+                {
+                    DungeonMetadata metadata = new DungeonMetadata();
+
+                    if (dungeonNode.Name == "dungeonRoom")
```

Comment:
you can invert this so it stays cleaner or use linq todo something like `document.DocumentElement.ChildNodes.Where(predicate)`